src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,30 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithFormSubmit extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(
+      ".modal__button_confirm"
+    );
+    this._submitButtonValue = this._submitButton.textContent;
+  }
+
+  setSubmitAction(action) {
+    this._submitAction = action;
+  }
+
+  setLoading(isLoading, buttonValue) {
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
Completed!